src/detectNode.js

```diff
@@ -0,0 +1,32 @@
+var isNodeEnvironment = function() {
+    // Same as in shell.js
+    var ENVIRONMENT_IS_WEB = false;
+    var ENVIRONMENT_IS_WORKER = false;
+    var ENVIRONMENT_IS_NODE = false;
+    var ENVIRONMENT_IS_SHELL = false;
+
+    // Three configurations we can be running in:
+    // 1) We could be the application main() thread running in the main JS UI thread. (ENVIRONMENT_IS_WORKER == false and ENVIRONMENT_IS_PTHREAD == false)
+    // 2) We could be the application main() thread proxied to worker. (with Emscripten -s PROXY_TO_WORKER=1) (ENVIRONMENT_IS_WORKER == true, ENVIRONMENT_IS_PTHREAD == false)
+    // 3) We could be an application pthread running in a worker. (ENVIRONMENT_IS_WORKER == true and ENVIRONMENT_IS_PTHREAD == true)
+
+    if (Module['ENVIRONMENT']) {
```

Comment:
`Module` may not exist here, first because we may have a different export name, second because we just have the modularize function, and it's parameter is the Module object, but it will only be called later.

I think all we can do is the code path that does not use `Module`. I think that should be enough here?